pkg/aws/awsmachinedeployment/validate_awsmachinedeployment.go

```diff
@@ -155,6 +197,26 @@ func (v *Validator) MachineDeploymentAnnotationPauseTimeIsValid(awsMachineDeploy
 	return nil
 }
 
+func (v *Validator) ValidateCluster(awsMachineDeployment infrastructurev1alpha2.AWSMachineDeployment) error {
+	var err error
+
+	// Retrieve the `Cluster` CR related to this object.
+	cluster, err := aws.FetchCluster(&aws.Handler{K8sClient: v.k8sClient, Logger: v.logger}, &awsMachineDeployment)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+	// make sure the cluster is not deleted
+	if cluster.DeletionTimestamp != nil {
+		v.logger.Log("level", "debug", "message", fmt.Sprintf("AWSMachineDeployment could not be created because Cluster '%s' is in deleting state.",
```

Comment:
I think this should be removed, either we log or we return the error.

Reply:
Makes sense. In that case I want to make sure that the error appears in the logs after validation failed.